### PR TITLE
DUOS-1292[risk=no] Chair Console: Update for DAR Collection

### PR DESCRIPTION
**Pull request by JVThomas**
Addresses [DUOS-1292](https://broadworkbench.atlassian.net/browse/DUOS-1292)

PR adds the Vote, Open, and Cancel buttons to the Chair Collection table for the Chair Console (on the `new_chair_console` route). 

Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/SimpleButton.js [hunk 1]
```diff
@@ -28,8 +29,8 @@ export default function SimpleButton(props) {
   const [style, setStyle] = useState({});
 
   useEffect(() => {
-    updateStyle('white', baseColor, additionalStyle, false, disabled);
-  }, [baseColor, additionalStyle, disabled]);
+    updateStyle(baseColor, fontColor, additionalStyle, false, disabled);
+  }, [baseColor, additionalStyle, disabled, fontColor]);
```
Comment (rushtong) [inline src/components/SimpleButton.js:33]: Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'updateStyle'. Either include it or remove the dependency array.](https://app.codacy.com/gh/DataBiosphere/duos-ui/pullRequest?prid=9165245)
```suggestion
  }, [baseColor, additionalStyle, disabled, fontColor, updateStyle]);
```

File: src/components/dar_collection_table/ChairActions.js [hunk 2]
```diff
@@ -26,44 +25,49 @@ const initUserData = ({dars, elections, relevantDatasets}) => {
...
       const datasetId = !isEmpty(dar.data) ? dar.data.datasetIds[0] : undefined;
-      return includes(datasetId)(relevantDatasetIds);
+      return includes(datasetId, relevantDatasetIds) ? isEmpty(dar.elections) : false;
     })(dars);
-    const relevantElections = filter((election) =>
-      includes(election.datasetId)(relevantDatasetIds)
-    )(elections);
-    return {relevantEmptyDars, relevantElections, relevantDatasetIds};
+    const relevantElections = filter((election) => {
+      //NOTE: not all elections have the dataSetId attribute tied to it (not sure why)
```
Comment (rushtong) [inline src/components/dar_collection_table/ChairActions.js:33]: Yes, this is something we need to look more closely into for dev/staging/prod. I dug through the db changelogs and we don't seem to have tried to back populate these values for older elections. So, we do need to be flexible in the UI until we address this on the database side.

File: src/pages/dar_collection_review/DarCollectionReview.js [hunk 3]
```diff
@@ -7,11 +7,10 @@ import TabControl from '../../components/TabControl';
 import RedirectLink from '../../components/RedirectLink';
 import ReviewHeader from './ReviewHeader';
 import ApplicationInformation from './ApplicationInformation';
-import { find, isEmpty, flow, filter, map, flatMap } from 'lodash/fp';
+import { find, isEmpty, flow, filter, map, flatMap, isNil } from 'lodash/fp';
 import { generatePreProcessedBucketData, processDataUseBuckets } from '../../utils/DarCollectionUtils';
 import DataUseVoteSummary from '../../components/common/DataUseVoteSummary/DataUseVoteSummary';
 import VotesPieChart from '../../components/common/VotesPieChart';
-import { isNil } from 'lodash';
```
Comment (rushtong) [inline src/pages/dar_collection_review/DarCollectionReview.js:14]: Nice cleanup

**Review by rushtong — APPROVED** [review]
This looks fantastic, thank you for the extensive tests 👍🏽

File: src/components/dar_collection_table/ChairActions.js (hunk 2 above)
Comment (rushtong) [reply src/components/dar_collection_table/ChairActions.js:33]: See https://broadworkbench.atlassian.net/browse/DUOS-1689

**Review by quazi-h — APPROVED** [review]
PR looks good and I observed the correct behavior on the chair console.

The Vote button was displayed for me for new DARs that were submitted.
I was only shown an Update button after submitting votes.
The review_results page is linked when I click on the status including Denied, Approved, and Unreviewed. When an election is still open, there is no link but I think that makes sense because there wouldn;t even be a review page for that DAR, right?

**Comment by JVThomas** [comment]
> PR looks good and I observed the correct behavior on the chair console.
> 
> The Vote button was displayed for me for new DARs that were submitted. I was only shown an Update button after submitting votes. The review_results page is linked when I click on the status including Denied, Approved, and Unreviewed. When an election is still open, there is no link but I think that makes sense because there wouldn;t even be a review page for that DAR, right?

I think you visited the current Chair Console. The PR makes a `new_chair_console` page to show off the new Chair Console. The PR's primarily the table styling, the new table, and the updated chair actions (Though the linkable status is a good idea).

**Review by quazi-h — COMMENTED** [review]
Code changes look good to me.
The new_chair_console page also looks good - I saw the Open, Vote, and Cancel buttons present where they should be. One difference I noticed though was that I have a certain DAC id:  439 (DAR-506) that has the "Update Vote" and "Cancel" options on the old chair_console but only has "Vote" and "Cancel" options on the new_console_chair page.

When I click on "Vote" option on the new_chair_console, it errors out:
Screen Shot 2022-04-05 at 9 27 56 AM

**Comment by JVThomas** [comment]
> Code changes look good to me. The new_chair_console page also looks good - I saw the Open, Vote, and Cancel buttons present where they should be. One difference I noticed though was that I have a certain DAC id: 439 (DAR-506) that has the "Update Vote" and "Cancel" options on the old chair_console but only has "Vote" and "Cancel" options on the new_console_chair page.
> 
> When I click on "Vote" option on the new_chair_console, it errors out: Screen Shot 2022-04-05 at 9 27 56 AM

Looks like that bug is happening on the `DarCollectionReview` component, not the console. Can you file this as a bug ticket outlining what happens and how to recreate the bug?

As for the Update Vote issue, that should be fixed now with the recent commits.

**Review by quazi-h — APPROVED** [review]
Changes look good, and I see the correct "Update Vote" button now instead of "Vote"

**Pull request merged**